backoff.go

```diff
@@ -0,0 +1,40 @@
+package pool
+
+import "time"
+
+const (
+	// maxBackOff is the maximum amount of that that we'll wait to retry to
+	// w/e operation we're wanting to back off.
+	maxBackOff = time.Minute
+
+	// minBackOff is the smallest amount of time we'll wait until we retry
+	// the given operation.
+	minBackOff = time.Millisecond * 100
+)
+
+// backOffer implements a simple exponential back off scheme using a channel
+// and a timer.
+type backOffer struct {
+	backOffPeriod time.Duration
+}
+
+// backOff returns a channel that should be used to wait for the current back
+// off period before doubling the period for the subsequent call to this
+// method. A quit channel should be specified to allow the back off to exit if
```

Comment:
Nit: comment about quit channel can be removed?

Reply:
Yep, was left over from an earlier design.